### PR TITLE
check milestone title uniqueness

### PR DESCRIPTION
**Pull request by mdehghani**
refs @giveth/giveth-dapp#1521

### CODE REVIEW

File: src/services/milestones/milestones.hooks.js [hunk 1]
```diff
@@ -330,6 +331,7 @@ module.exports = {
     create: [
       checkConversionRates(),
       checkMilestoneDates(),
+      checkMilestoneName(),
```
Comment (aminlatifi) [inline src/services/milestones/milestones.hooks.js:334]: We should consider patch and update too, use may edit milestone after he has created.

**Review by aminlatifi — CHANGES_REQUESTED** [review]

File: src/services/milestones/checkMilestoneName.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  if (!data.campaignId) {
+    const milestone = await milestoneService.get(context.id);
+    if (milestone) {
+      data.campaignId = milestone.campaignId;
+    }
+  }
+  const milestoneWithSameName = await milestoneService.find({
+    query: {
+      campaignId: data.campaignId,
+      title: new RegExp(`\\s*${title.replace(/[.*+?^${}()|[\]\\]/g, '\\$&')}\\s*`),
```
Comment (aminlatifi) [inline src/services/milestones/checkMilestoneName.js:22]: I am not sure whether it's a good idea to match it case insensitive. @mdehghani what do you think?

**Review by aminlatifi — CHANGES_REQUESTED** [review]
I tested this PR to create two milestones with identical name. I created the second one while the first one wasn't mined (was in pending state). There are two issues in my opinion:
1. The first one remains in pending state for ever. I don't think it's related to creation of second milestone, since the processing `ProjectAdded` event patches the milestone and has title in addition to status in the mutation query. One solution would be adding the `_id: {$ne: milestone.id}` to the query, @mdehghani you would find better way.
2. UI is not ready, the tx popup is shown and after user creates the tx and it's mined Dapp says milestone is created. @mdehghani we can make this PR as draft till the appropriate PR is created for Dapp.

**Comment by mdehghani** [comment]
UI has been changed previously. You can find the related PR here:
https://github.com/Giveth/giveth-dapp/pull/1728

File: src/services/milestones/checkMilestoneName.js (hunk 2 above)
Comment (mdehghani) [reply src/services/milestones/checkMilestoneName.js:22]: It's not matching case insensitive. It tries to trim the title (remove whitespaces from the beginning and the end of the title).
But it seems there is a problem in the regex. I will change it to a simpler regex.

**Pull request merged**